application/controllers/ScheduleController.php

```diff
@@ -51,10 +54,23 @@ public function indexAction(): void
             ->setAction(Url::fromRequest()->getAbsoluteUrl())
             ->populate(['mode' => $this->params->get('mode')])
             ->on(Form::ON_SUCCESS, function (ScheduleDetail\Controls $controls) use ($id) {
-                $this->redirectNow(Links::schedule($id)->with(['mode' => $controls->getMode()]));
+                $redirectUrl = Links::schedule($id)->with(['mode' => $controls->getMode()]);
+                $requestUrl = Url::fromRequest();
+                if ($requestUrl->getParam('mode') !== $controls->getValue('mode')) {
```

Comment:
Why…? If you wanted to avoid the redirect although nothing changed, then this is the wrong place, and PR.